src/garage/_dtypes.py

```diff
@@ -230,9 +236,23 @@ def __new__(cls, env_spec, observations, last_observations, actions,
                 'step_types tensor must be dtype `StepType`, but got tensor '
                 'of dtype {} instead.'.format(step_types.dtype))
 
-        return super().__new__(EpisodeBatch, env_spec, observations,
-                               last_observations, actions, rewards, env_infos,
-                               agent_infos, step_types, lengths)
+        # episode_infos
+        for key, val in episode_infos.items():
+            if not isinstance(val, (dict, np.ndarray)):
+                raise ValueError(
+                    'Each entry in episode_infos must be a numpy array or '
```

Comment:
according to the docstring the items can only be numpy arrays -- why allow dicts?